neo4j/src/main/java/org/hibernate/ogm/datastore/neo4j/dialect/impl/BaseNeo4jQueries.java

```diff
@@ -64,31 +65,51 @@ protected static void appendProperties(StringBuilder queryBuilder, String[] colu
 	 *
 	 * MATCH (owner:ENTITY:table {id: {0}})
 	 */
-	protected static void appendMatchOwnerEntityNode(StringBuilder queryBuilder, EntityKeyMetadata ownerEntityKeyMetadata) {
+	protected static void appendMatchOwnerEntityNode(StringBuilder queryBuilder, EntityKeyMetadata ownerEntityKeyMetadata, TupleTypeContext tupleTypeContext) {
 		queryBuilder.append( "MATCH " );
-		appendEntityNode( "owner", ownerEntityKeyMetadata, queryBuilder );
+		appendEntityNode( "owner", ownerEntityKeyMetadata, tupleTypeContext, queryBuilder );
 	}
 
 	/*
 	 * Example:
 	 *
 	 * (owner:ENTITY:table {id: {0}})
 	 */
-	protected static void appendEntityNode(String alias, EntityKeyMetadata entityKeyMetadata, StringBuilder queryBuilder) {
-		appendEntityNode( alias, entityKeyMetadata, queryBuilder, 0 );
+	protected static void appendEntityNode(String alias, EntityKeyMetadata entityKeyMetadata, TupleTypeContext tupleTypeContext, StringBuilder queryBuilder) {
+		appendEntityNode( alias, entityKeyMetadata, tupleTypeContext, queryBuilder, 0, false );
 	}
 
-	protected static void appendEntityNode(String alias, EntityKeyMetadata entityKeyMetadata, StringBuilder queryBuilder, int offset) {
+	/*
+	 * At the moment, we want to add the discriminator value as a label in Neo4j, but we still want to keep compatibility with the old mapping.
+	 * For this reason we don't to consider the discriminator label when executing a select.
+	 * <p>
+	 * We are going to change the mapping in the next major release.
+	 */
+	protected static void appendEntityNodeWithDiscriminatorLabel(String alias, EntityKeyMetadata entityKeyMetadata, TupleTypeContext tupleTypeContext, StringBuilder queryBuilder) {
+		appendEntityNode( alias, entityKeyMetadata, tupleTypeContext, queryBuilder, 0, true );
+	}
+
+	protected static void appendEntityNode(String alias, EntityKeyMetadata entityKeyMetadata, TupleTypeContext tupleTypeContext, StringBuilder queryBuilder, int offset, boolean addDiscriminatorLabel) {
 		queryBuilder.append( "(" );
 		queryBuilder.append( alias );
 		queryBuilder.append( ":" );
 		queryBuilder.append( ENTITY );
 		queryBuilder.append( ":" );
 		appendLabel( entityKeyMetadata, queryBuilder );
+		if ( addDiscriminatorLabel ) {
+			appendDiscriminatorLabels( tupleTypeContext, queryBuilder );
+		}
 		appendProperties( queryBuilder, entityKeyMetadata.getColumnNames(), offset );
 		queryBuilder.append( ")" );
 	}
 
+	public static void appendDiscriminatorLabels(TupleTypeContext tupleTypeContext, StringBuilder queryBuilder) {
```

Comment:
s/appendDiscriminatorLabels/appendDiscriminatorLabel/
(there is only one label)